.scripts/util.sh

```diff
@@ -38,85 +38,88 @@ copy_modules() {
-	cp "$SRC/sys/sockets/xif/dial.xif" "$MCHDIR/dial.xix"
-	cp "$SRC/sys/sockets/xif/pamsdma.xif" "$MCHDIR/pamsdma.xix"
-	cp "$SRC/sys/sockets/xif/rtl8012st.xif" "$MCHDIR/rtl8012st.xix"
-	cp "$SRC/sys/xdd/mfp/mfp.xdd" "$MCHDIR"
+	cp "$SRC/sys/sockets/xif/biodma_000.xif" "$MCHDIR/biodma_000.xix"
+	cp "$SRC/sys/sockets/xif/de600_000.xif" "$MCHDIR/de600.xix"
+	cp "$SRC/sys/sockets/xif/dial_000.xif" "$MCHDIR/dial.xix"
+	cp "$SRC/sys/sockets/xif/pamsdma_000.xif" "$MCHDIR/pamsdma.xix"
+	cp "$SRC/sys/sockets/xif/rtl8012st_000.xif" "$MCHDIR/rtl8012st.xix"
+	cp "$SRC/sys/xdd/mfp/.compile_000/mfp.xdd" "$MCHDIR"
 }
 copy_megast_modules() {
 	local MCHDIR="$1/megast"
 	mkdir -p "$MCHDIR"
-	cp "$SRC/sys/sockets/xif/lance.xif" "$MCHDIR/lance.xix"
-	cp "$SRC/sys/sockets/xif/rieblmst.xif" "$MCHDIR/riebl.xix"
-	cp "$SRC/sys/sockets/xif/rieblmst_fast.xif" "$MCHDIR/riebl_fast.xix"
-	cp "$SRC/sys/sockets/xif/rieblspc.xif" "$MCHDIR/rieblspc.xix"
-	cp "$SRC/sys/sockets/xif/rieblspc_fast.xif" "$MCHDIR/riebls_fast.xix"
-	cp "$SRC/sys/sockets/xif/rtl8012st.xif" "$MCHDIR/rtl8012st.xix"
-	cp "$SRC/sys/xdd/mfp/mfp.xdd" "$MCHDIR"
+	cp "$SRC/sys/sockets/xif/lance_000.xif" "$MCHDIR/lance.xix"
+	cp "$SRC/sys/sockets/xif/rieblmst_000.xif" "$MCHDIR/riebl.xix"
+	cp "$SRC/sys/sockets/xif/rieblmst_fast_000.xif" "$MCHDIR/riebl_fast.xix"
+	cp "$SRC/sys/sockets/xif/rieblspc_000.xif" "$MCHDIR/rieblspc.xix"
+	cp "$SRC/sys/sockets/xif/rieblspc_fast_000.xif" "$MCHDIR/riebls_fast.xix"
+	cp "$SRC/sys/sockets/xif/rtl8012st_000.xif" "$MCHDIR/rtl8012st.xix"
+	cp "$SRC/sys/xdd/mfp/.compile_000/mfp.xdd" "$MCHDIR"
 }
 copy_ste_modules() {
 	local MCHDIR="$1/ste"
 	mkdir -p "$MCHDIR"
-	cp "$SRC/sys/sockets/xif/rtl8012st.xif" "$MCHDIR/rtl8012st.xix"
-	cp "$SRC/sys/xdd/mfp/mfp.xdd" "$MCHDIR"
+	cp "$SRC/sys/sockets/xif/rtl8012st_000.xif" "$MCHDIR/rtl8012st.xix"
+	cp "$SRC/sys/xdd/mfp/.compile_000/mfp.xdd" "$MCHDIR"
 }
 copy_megaste_modules() {
 	local MCHDIR="$1/megaste"
 	mkdir -p "$MCHDIR"
-	cp "$SRC/sys/sockets/xif/rieblste.xif" "$MCHDIR/rieblste.xix"
-	cp "$SRC/sys/sockets/xif/rtl8012st.xif" "$MCHDIR/rtl8012st.xix"
-	cp "$SRC/sys/xdd/mfp/mfp.xdd" "$MCHDIR"
-	cp "$SRC/sys/xdd/scc/scc.xdd" "$MCHDIR"
+	cp "$SRC/sys/sockets/xif/rieblste_000.xif" "$MCHDIR/rieblste.xix"
+	cp "$SRC/sys/sockets/xif/rtl8012st_000.xif" "$MCHDIR/rtl8012st.xix"
+	cp "$SRC/sys/xdd/mfp/.compile_000/mfp.xdd" "$MCHDIR"
+	cp "$SRC/sys/xdd/scc/.compile_000/scc.xdd" "$MCHDIR"
 }
 copy_tt_modules() {
 	local MCHDIR="$1/tt"
+	local TARGET="$2"
 	mkdir -p "$MCHDIR"
-	cp "$SRC/sys/sockets/xif/daynaport/scsilink.xif" "$MCHDIR/scsilink.xix"
-	cp "$SRC/sys/sockets/xif/lance.xif" "$MCHDIR/lance.xix"
-	cp "$SRC/sys/sockets/xif/riebltt.xif" "$MCHDIR/riebltt.xix"
-	cp "$SRC/sys/sockets/xif/rtl8012.xif" "$MCHDIR/rtl8012.xix"
-	cp "$SRC/sys/xdd/mfp/mfp.xdd" "$MCHDIR"
-	cp "$SRC/sys/xdd/scc/scc.xdd" "$MCHDIR"
+	cp "$SRC/sys/sockets/xif/daynaport/.compile_${TARGET}/scsilink.xif" "$MCHDIR/scsilink.xix"
```

Comment:
Why is now ${TARGET} used and not hardcoded _030 (similar to _000 above) ? TT can't be run on other than 030 CPU.

Reply:
Because it is called with different CPU targets (030 for tt, 02060 for falcon)

Reply:
True but those are copied into the MCHDIR, i.e. being used just on that machine. In this case the TT.

When I introduced the 02060 target, the idea was to offer a package which includes these CPU families (i.e. one archive to run on 020, 030, 040, 060), not to explicitly compile everything for that CPU (of course, it makes sense for the tools and common code because you don't know on which machine it will end up running).

So it's totally fine to keep `copy_tt_modules` as copying, well, modules aimed for the TT, i.e. 030 CPU.

Reply:
Yes, but it currently does not make much difference anyway. There is only one archive for tt-falcon, not separate ones, so these modules will be overwritten later by copy_falcon_modules.

Reply:
That's my point, they won't -- cooy_falcon_modules will copy stuff into Falcon MCHDIR so there's really no point in compiling TT modules for other than 030.

Reply:
But this already happens in other places too: https://github.com/freemint/freemint/actions/runs/10158122454/job/28089528657?pr=348#step:8:5308

And IIRC, i got some errors from cp command when i omit 02060

Reply:
usb modules are OK to compile for 02060: that's the common code I was referring to. With those, you can't tell whether the zip will be unpacked/booted on the TT or Falcon, so they are 020-60. However TT, Falcon, Aranym MCHDIRs are solely for given machine, there's no benefit in compiling them for something else.

Actually your work here is very beneficial: until now we didn't have an option to compile, say, mfp.xdd for 030 and for 040 and now we can do it per-MCHDIR: Falcon's and TT's MCHDIR would contain 030 mfp.xdd, Aranym 040 mfp.xdd, Hades 020-60 mfp.xdd ... very cool.

So I think this is worth looking into, TT should be definitely 030 only.

Reply:
Yes, i've changed that copy_tt_modules already to use 030, not `$TARGET` (not pushed yet). I first thought those modules would be overwritten by copy_falcon_modules, until i realized that they use different directories.

What should be checked also is copy_m68k_modules where i did a similar change. Until now some of these modules were only compiled for `000`.

Reply:
On the contrary, the `copy_m68k_modules` change is exactly what we want: each target will get optimised modules. In case of the bootable TT/Falcon archive, we get 020-60 modules instead of 000 modules => hopefully better performance.

There used to be an issue when actually USB modules were compiled for 000 only and tested on real (030/060) hardware and after changing the CPU flags timing didn't match anymore. However that was fixed by @DavidGZ a long time ago. Hopefully other modules (especially the new ones: asix, picowifi) don't use constructs like "move #random_constant, d0; dbra d0, loop" for delays.